unison-cli/src/Unison/Codebase/Editor/HandleInput/Pull.hs

```diff
@@ -223,16 +224,19 @@ loadRemoteNamespaceIntoMemory syncMode pullMode remoteNamespace = do
         Cli.returnEarly (Output.GitError err)
     ReadShare'LooseCode repo -> loadShareLooseCodeIntoMemory repo
     ReadShare'ProjectBranch remoteBranch -> do
-      downloadShareProjectBranch remoteBranch
-      let causalHash = Common.hash32ToCausalHash (Share.hashJWTHash (remoteBranch ^. #branchHead))
+      projectBranchCausalHashJWT <- downloadShareProjectBranch (pullMode == Input.PullWithoutHistory) remoteBranch
+      let causalHash = Common.hash32ToCausalHash (Share.hashJWTHash projectBranchCausalHashJWT)
       liftIO (Codebase.expectBranchForHash codebase causalHash)
 
 -- | @downloadShareProjectBranch branch@ downloads the given branch.
-downloadShareProjectBranch :: Share.RemoteProjectBranch -> Cli ()
-downloadShareProjectBranch branch = do
+downloadShareProjectBranch :: Bool -> Share.RemoteProjectBranch -> Cli HashJWT
+downloadShareProjectBranch useSquashedIfAvailable branch = do
+  let remoteProjectBranchName = branch ^. #branchName
   let repoInfo = Share.RepoInfo (into @Text (ProjectAndBranch (branch ^. #projectName) remoteProjectBranchName))
-      causalHashJwt = branch ^. #branchHead
-      remoteProjectBranchName = branch ^. #branchName
+  let causalHashJwt =
+        if useSquashedIfAvailable
+          then (branch ^. #branchHead)
+          else fromMaybe (branch ^. #branchHead) (branch ^. #squashedBranchHead)
```

Comment:
According to how the server-side is implemented this is impossible, but if the squashed head isn't set it will just download the main head instead rather than erroring. I can swap this with a returnEarly if anyone feels strongly about it one way or the other.

Reply:
Seems ok. Should we print a note and ask the user to let us know, then proceed to the slower download?

Reply:
I agree that we should either fail ("can't pull without history") or let the user know ("falling back on pulling with history"), but if we don't want to do either, let's at least add a comment reminding us that that's the plan

Reply:
Sounds good, it shouldn't ever happen and we'll want to know if it does, so I just swapped with with a call to `error`.